js/forum/src/components/EditUserModal.js

```diff
@@ -100,6 +111,24 @@ export default class EditUserModal extends Modal {
     );
   }
 
+  activate() {
+    this.loading = true;
+    const data = {
+      username: this.username(),
+      isActivated: true,
+    };
+    this.props.user.save(data, {errorHandler: this.onerror.bind(this)})
+      .then(() => {
+        this.isActivated(true);
+        this.loading = false;
+        m.redraw();
+      })
```

Comment:
Well init is only run when the model is opened. So this.isActivated is set to the activation status when it's opened. When the user is saved, it updates this.props.user but not this.isActivated.